res/database/Default/user/jorgaeff.xml

```diff
@@ -1440,6 +1441,7 @@
 				<all>${.csv:theprobablymost_csv:0}</all>
 			</description>
 			<variables>
+				<currentyear>${.worldtime:"year"}</currentyear>
```

Comment:
Das wäre ein Kandidat für die globalen Variablen, oder?

Reply:
Verstehe nicht wie / wo?

Reply:
Da nun globale Variablen funktionieren sollten, könnte `<currentyear>${.worldtime:"year"}</currentyear>` in en.xml definiert werden und dann mit `${currentyear}` überall direkt referenziert werden. Man spart sich die duplizierten lokalen Definitionen. Keine Änderungsaufforderung.

Reply:
Hätte jetzt gedacht dann wird sie nur einmal aufgelöst. Beim ersten Zugriff. Was ja nicht gewollt wäre.

Reply:
Derzeit werden die immer wieder neu aufgeloest.

Ich hatte zwar angefragt, ob die "resolved" Globalen (also bei denen die Expression ausgewertet wurde) jeweils zu den "resolved" Variablen eines Objektes gespeichert werden koennten, aber das ist ersteinmal nicht umgesetzt.
Vorteil dort waere, dass ein Elternelement eine Globale auswertet und Kinderelemente (Episoden bspweise) diese dann abrufen, statt neu auszuwerten. Das wuerde in diesem Fall bedeuten, dass "currentyear" nur einmalig pro Serie ausgewertet _wuerde_, sobald im Serienkopf `${currentyear}" zum Einsatz kaeme (bspweise in der Serienbeschreibung).

Dies bedeutet nun im Umkehrschluss: sobald wir das so machen, dass globale Variablen _immer_ neu ausgewertet werden, wir da das wie von Nittka vorgeschlagen machen koennen (currentyear-variable). Dann aber nicht mehr davon abweichen sollten (@nittka muesste einen entsprechenden Codehinweis hinterlegen ... bzw "Codewarnung").

Und es macht es in der Hinsicht etwas hmm "undeutlich", wann eine Variable "immer wieder" ausgewertet wird, und wann nur einmalig.
Kurzum: Man muss schauen, ob man die Varable bspweise vom Serienkopf erbt, oder von "global vars".

@nittka - ebenfalls interessant waere, ob man in einem Serienkopf folgendes machen kann (wenn currentyear bereits als global definiert ist)
```
<currentyear>${currentyear}</currentyear>
```
Denn nun wuerde currentyear als resolved variable im Kopf hinterlegt werden und Episoden wuerden _diese_ Variable nutzen, statt der globalen. Kann sinnvoll sein (globale "statisch" machen) aber ist natuerlich ein weiterer potentieller Stolperstein.

Klingt bloed -- aber vielleicht sollten "globale" mit einem "g" geprefixt werden oder aehnliches?

Reply:
um wirklich klar zu sein müsste es eigentlich eine art lambda-syntax geben, der klar macht, ich referenziere hier eine funktion, nicht einen aufgelösten wert.

aber `${.worldtime:"year"}` ist ja eigentlich kurz genug, ich referenzierte das lokal wirklich nur raus, weil ich dachte, das ist dann "safer" bzgl eventuellen "race conditions".

Reply:
Bitte den Initialkommentar erstmal einfach vergessen.
Globale Variablen und "Templatevariablen" (Variablen, die in einem Template definiert werden) sind ähnlich, verhalten sich aber definitiv nicht gleich.
Bevor weitreichende Designentscheidungen getroffen werden, sollten sie ausschließlich für feste Texte verwendet werden. Es ist vorbereitet, dass sie selbst wieder Ausdrücke enthalten können, aber davon sollte nur mit Vorsicht Gebrauch gemacht werden.